src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java

```diff
@@ -687,6 +687,29 @@ public boolean setPassenger(@NotNull Entity passenger)
 		return Collections.unmodifiableList(this.passengers);
 	}
 
+	/**
+	 * Returns all passengers, and passengers of passengers.
+	 *
+	 * @return All indirect passengers.
+	 */
+	public @NotNull List<Entity> getIndirectPassengers()
```

Comment:
The name would imply that this would not include direct passengers.
The term you are looking for is "transitive" passengers.
Perhaps it would be better to just add an overloaded method `getPassengers(boolean transitive)`.

If true, then return the list as described, if not then fallback to #getPassengers()

The javadocs here should also state that the returned list is ~~immutable~~ unmodifiable.